### PR TITLE
Change character type to uint8_t

### PR DESCRIPTION
**Pull request by seanshpark**
This commit is to apply latest character type change in JerryScript as uint8_t
- Update to latest JerryScript
- add new jschar type as uint8_t and change char types to this
- disable JObject(const char*) constructor which makes problem
- update JHandlerType for new jerry_api_length_t

IoT.js-DCO-1.0-Signed-off-by: SaeHie Park saehie.park@samsung.com

Related issue #118

### CODE REVIEW

File: src/iotjs_util.cpp [hunk 1]
```diff
@@ -45,19 +65,19 @@ char* ReadFile(const char* path) {
 }
 
 
-char* AllocBuffer(size_t size) {
-  char* buff = static_cast<char*>(malloc(size));
+jschar* AllocBuffer(size_t size) {
```
Comment (ILyoan) [inline src/iotjs_util.cpp]: `AllocBuffer` is not for creating string buffer, it is for creating byte buffer(creating `size` of octets). Thus changed prototype is not correctly describe the purpose of this function.

File: src/iotjs_util.cpp [hunk 2]
```diff
@@ -45,19 +65,19 @@ char* ReadFile(const char* path) {
...
+  jschar* buff = static_cast<jschar*>(malloc(size));
   memset(buff, 0, size);
   return buff;
 }
 
 
-char* ReallocBuffer(char* buffer, size_t size) {
-  return static_cast<char*>(realloc(buffer, size));
+jschar* ReallocBuffer(jschar* buffer, size_t size) {
```
Comment (ILyoan) [inline src/iotjs_util.cpp]: Hero also.

File: src/iotjs_util.cpp [hunk 3]
```diff
@@ -45,19 +65,19 @@ char* ReadFile(const char* path) {
...
+  return static_cast<jschar*>(realloc(buffer, size));
 }
 
 
-void ReleaseBuffer(char* buffer) {
+void ReleaseBuffer(jschar* buffer) {
```
Comment (ILyoan) [inline src/iotjs_util.cpp]: Here also

File: src/iotjs_types.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ */
+
+
+#ifndef IOTJS_TYPES_H
+#define IOTJS_TYPES_H
+
+
+typedef uint8_t jschar;  /* follow jerry_api_char_t type which is uint8_t */
+
+#define _JSCT (jschar*)
```
Comment (ILyoan) [inline src/iotjs_types.h]: Looks like this macro is tempting writing codes which does not look follow C grammar. for example
```
foo(_JSCT iotjs);
bar(_JSCT "bar");
```

What about function like macro such as:
```
#define JSCT(a) static_cast<jschar*>(a)

...

foo(JSCT(iotjs));
bar(JSCT("bar"));
```

File: src/iotjs_module_buffer.h [hunk 5]
```diff
@@ -42,16 +42,16 @@ class BufferWrap : public JObjectWrap {
...
   int Compare(const BufferWrap& other) const;
 
-  size_t Copy(char* src, size_t len);
-  size_t Copy(char* src, size_t src_from, size_t src_to, size_t dst_from);
+  size_t Copy(jschar* src, size_t len);
+  size_t Copy(jschar* src, size_t src_from, size_t src_to, size_t dst_from);
 
  protected:
-  char* _buffer;
+  jschar* _buffer;
```
Comment (ILyoan) [inline src/iotjs_module_buffer.h]: `Buffer` is for array of bytes, not for javascript string. Thus `_buffer` should be a type representing array of bytes.

File: src/iotjs_binding.h [hunk 6]
```diff
@@ -150,10 +151,10 @@ class JObject {
   // Returns pontiner to null terminated string contents of string object.
   // Returned pointer should be released using `ReleaseCString()` when it become
   // unnecessary.
-  char* GetCString();
+  jschar* GetCString();
```
Comment (ILyoan) [inline src/iotjs_binding.h]: This function should return type of C String not javascript string.

Comment (ILyoan) [inline src/iotjs_binding.h]: Or I think we should rename it

File: src/iotjs_types.h (hunk 4 above)
Comment (seanshpark) [inline src/iotjs_types.h]: Ok, looks better

File: src/iotjs_binding.h (hunk 6 above)
Comment (seanshpark) [inline src/iotjs_binding.h]: Rename GetCString() to GetByteString() seems nice.

**Comment by ILyoan** [comment]
@seanshpark  
I think we need to introduce kind of String wrapper.
Before it, let's go with this.

Let's make it merged

**Comment by seanshpark** [comment]
OK

**Pull request merged**